cpp/src/arrow/acero/CMakeLists.txt

```diff
@@ -29,7 +29,11 @@ endmacro()
 
 set(ARROW_ACERO_SRCS
     accumulation_queue.cc
-    aggregate_node.cc
```

Comment:
`aggregate_node.cc` was split into `scalar_aggregate_node.cc`, `groupby_aggregate_node.cc` and `aggregate_internal.cc` - the existing code was moved as-is